# Yeni Arkadaşımız "EDA"

![](https://miro.medium.com/max/640/1*cBOu28eaQKCTN-y6k5imrw.png)


**Bu fotoğraf "Midjourney" tarafından üretildi.**

Bu notebook,***bu medium yazısının*** notebook halidir.

## İçindekiler:
- Neden "EDA" bizim için önemli?
- Hadi EDA dediğimiz şey ne onu öğrenelim!
- .head()
- .tail()
- .sample()
- .describe()
- .corr()
- .info()
- .isnull().sum()
- .shape


---

Bu yazı ile ismi "**EDA**" olan yeni bir arkadaş edinceksiniz. Veri Bilimcilerin en yakın arkadaşlarından biridir **EDA**.


Hadi sizleri **EDA** ile tanıştırıyım. **EDA**, Türkçemizde "***Açınsayıcı Veri Çözümlemesi***" veya "***Keşifsel Veri Analizi***" diye çevrilmiş olup İngilizce aslı "***Exploratory Data Analysis***" dir.

![](https://miro.medium.com/max/1400/0*SMTyS7nGrMze2Qxz)

[Unsplash](https://unsplash.com/?utm_source=medium&utm_medium=referral)'de [Natalya Khartukova](https://unsplash.com/@nataxart?utm_source=medium&utm_medium=referral) tarafından bir fotoğraf

> # Neden "EDA" bizim için önemli?

Verileri okuduktan sonra verilerden bir anlam çıkaramazsak bir şey yapamayız. Dolayısıyla verileri açıkça görmek, verileri daha iyi anlamak verileri temizlemek için **EDA**'ya çok ihtiyacımız var

># Hadi EDA dediğimiz şey ne onu öğrenelim!

Bu yazı için **Titanic veri setini** kullanıcam.

![](https://miro.medium.com/max/828/0*9LrHADi8ECdGOwFJ)

[Unsplash](https://unsplash.com/?utm_source=medium&utm_medium=referral)'de [Nicolas Hoizey](https://unsplash.com/@nhoizey?utm_source=medium&utm_medium=referral) tarafından bir fotoğraf

> # .head()

**head()** fonksiyonu verimizdeki ilk 5 satırı gösterir, fakat eğer içine bir sayı girerseniz o sayıya kadar olan satırları gösterir. Mesela **.head(15)** ilk 15 satırı gösterir.

Eğer veri nasıl okunur bilmiyorsanız [bu yazıyı](https://medium.com/@ahmettalhabektas_tr/veri-d%C3%BCnyas%C4%B1na-i%CC%87lk-ad%C4%B1mlar-3044f0d42170) veya  [bu kodu](https://www.kaggle.com/code/ahmettalhabektastr/veri-d-nyas-na-lk-ad-mlar) okumalısınız.

### **Hadi örnek üstünde yapalım!**

In [1]:
import pandas as pd 
df=pd.read_csv("/kaggle/input/titanic/train.csv")
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [2]:
df.head(15)

PassengerId  Survived  Pclass  \
0             1         0       3   
1             2         1       1   
2             3         1       3   
3             4         1       1   
4             5         0       3   
5             6         0       3   
6             7         0       1   
7             8         0       3   
8             9         1       3   
9            10         1       2   
10           11         1       3   
11           12         1       1   
12           13         0       3   
13           14         0       3   
14           15         0       3   

                                                 Name     Sex   Age  SibSp  \
0                             Braund, Mr. Owen Harris    male  22.0      1   
1   Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                              Heikkinen, Miss. Laina  female  26.0      0   
3        Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                            Allen, Mr. William Henry    male  35.0      0   
5                                    Moran, Mr. James    male   NaN      0   
6                             McCarthy, Mr. Timothy J    male  54.0      0   
7                      Palsson, Master. Gosta Leonard    male   2.0      3   
8   Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                 Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   
10                    Sandstrom, Miss. Marguerite Rut  female   4.0      1   
11                           Bonnell, Miss. Elizabeth  female  58.0      0   
12                     Saundercock, Mr. William Henry    male  20.0      0   
13                        Andersson, Mr. Anders Johan    male  39.0      1   
14               Vestrom, Miss. Hulda Amanda Adolfina  female  14.0      0   

    Parch            Ticket     Fare Cabin Embarked  
0       0         A/5 21171   7.2500   NaN        S  
1       0          PC 17599  71.2833   C85        C  
2       0  STON/O2. 3101282   7.9250   NaN        S  
3       0            113803  53.1000  C123        S  
4       0            373450   8.0500   NaN        S  
5       0            330877   8.4583   NaN        Q  
6       0             17463  51.8625   E46        S  
7       1            349909  21.0750   NaN        S  
8       2            347742  11.1333   NaN        S  
9       0            237736  30.0708   NaN        C  
10      1           PP 9549  16.7000    G6        S  
11      0            113783  26.5500  C103        S  
12      0         A/5. 2151   8.0500   NaN        S  
13      5            347082  31.2750   NaN        S  
14      0            350406   7.8542   NaN        S

> # .tail()

**tail()** fonksiyonu **head()** fonksiyonuna çok benzer, fakat tersten satırları gösterir. Yani içine sayı girilmezse son 5 satır girlirse sondan o satıra kadar gösterir.

### **Hadi örnek üstünde yapalım!**

In [3]:
df.tail()

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

In [4]:
df.tail(10)

PassengerId  Survived  Pclass                                      Name  \
881          882         0       3                        Markun, Mr. Johann   
882          883         0       3              Dahlberg, Miss. Gerda Ulrika   
883          884         0       2             Banfield, Mr. Frederick James   
884          885         0       3                    Sutehall, Mr. Henry Jr   
885          886         0       3      Rice, Mrs. William (Margaret Norton)   
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
881    male  33.0      0      0            349257   7.8958   NaN        S  
882  female  22.0      0      0              7552  10.5167   NaN        S  
883    male  28.0      0      0  C.A./SOTON 34068  10.5000   NaN        S  
884    male  25.0      0      0   SOTON/OQ 392076   7.0500   NaN        S  
885  female  39.0      0      5            382652  29.1250   NaN        Q  
886    male  27.0      0      0            211536  13.0000   NaN        S  
887  female  19.0      0      0            112053  30.0000   B42        S  
888  female   NaN      1      2        W./C. 6607  23.4500   NaN        S  
889    male  26.0      0      0            111369  30.0000  C148        C  
890    male  32.0      0      0            370376   7.7500   NaN        Q

> # .sample()

**sample()** fonksiyonu rastgele bir satır getirir.Fakat içine bir sayı yazarsanız o sayı kadar rastgele satır getirir.

### **Hadi örnek üstünde yapalım!**

In [5]:
df.sample()

PassengerId  Survived  Pclass                    Name   Sex   Age  SibSp  \
864          865         0       2  Gill, Mr. John William  male  24.0      0   

     Parch  Ticket  Fare Cabin Embarked  
864      0  233866  13.0   NaN        S

In [6]:
df.sample(15)

PassengerId  Survived  Pclass  \
69            70         0       3   
48            49         0       3   
481          482         0       2   
20            21         0       2   
631          632         0       3   
572          573         1       1   
537          538         1       1   
18            19         0       3   
594          595         0       2   
557          558         0       1   
782          783         0       1   
277          278         0       2   
294          295         0       3   
246          247         0       3   
149          150         0       2   

                                                  Name     Sex   Age  SibSp  \
69                                   Kink, Mr. Vincenz    male  26.0      2   
48                                 Samaan, Mr. Youssef    male   NaN      2   
481                   Frost, Mr. Anthony Wood "Archie"    male   NaN      0   
20                                Fynney, Mr. Joseph J    male  35.0      0   
631                        Lundahl, Mr. Johan Svensson    male  51.0      0   
572                   Flynn, Mr. John Irwin ("Irving")    male  36.0      0   
537                                LeRoy, Miss. Bertha  female  30.0      0   
18   Vander Planke, Mrs. Julius (Emelia Maria Vande...  female  31.0      1   
594                            Chapman, Mr. John Henry    male  37.0      1   
557                                Robbins, Mr. Victor    male   NaN      0   
782                             Long, Mr. Milton Clyde    male  29.0      0   
277                        Parkes, Mr. Francis "Frank"    male   NaN      0   
294                                   Mineff, Mr. Ivan    male  24.0      0   
246              Lindahl, Miss. Agda Thorilda Viktoria  female  25.0      0   
149                  Byles, Rev. Thomas Roussel Davids    male  42.0      0   

     Parch       Ticket      Fare Cabin Embarked  
69       0       315151    8.6625   NaN        S  
48       0         2662   21.6792   NaN        C  
481      0       239854    0.0000   NaN        S  
20       0       239865   26.0000   NaN        S  
631      0       347743    7.0542   NaN        S  
572      0     PC 17474   26.3875   E25        S  
537      0     PC 17761  106.4250   NaN        C  
18       0       345763   18.0000   NaN        S  
594      0  SC/AH 29037   26.0000   NaN        S  
557      0     PC 17757  227.5250   NaN        C  
782      0       113501   30.0000    D6        S  
277      0       239853    0.0000   NaN        S  
294      0       349233    7.8958   NaN        S  
246      0       347071    7.7750   NaN        S  
149      0       244310   13.0000   NaN        S

> # .describe()

**describe()** fonksiyonu istatiksel değerleri hesaplar (min, max, mean…) ve bu hesapladıkları değerleri gösterir.

### **Hadi örnek üstünde yapalım!**

In [7]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

> # .corr()

Her şeyin birbiriyle bir ilişkisi vardır. Hatta sizin bu yazıyı okumanız ile yarın sabah kahvaltınızda yumurta yemenizin bile bir bağlantısı var (En azından matematşksel olarak 😅). **corr()** fonksiyonu sütunların birbiri ile olan korelasyonunu bizlere gösterir.

### **Hadi örnek üstünde yapalım!**

In [8]:
df.corr()

PassengerId  Survived    Pclass       Age     SibSp     Parch  \
PassengerId     1.000000 -0.005007 -0.035144  0.036847 -0.057527 -0.001652   
Survived       -0.005007  1.000000 -0.338481 -0.077221 -0.035322  0.081629   
Pclass         -0.035144 -0.338481  1.000000 -0.369226  0.083081  0.018443   
Age             0.036847 -0.077221 -0.369226  1.000000 -0.308247 -0.189119   
SibSp          -0.057527 -0.035322  0.083081 -0.308247  1.000000  0.414838   
Parch          -0.001652  0.081629  0.018443 -0.189119  0.414838  1.000000   
Fare            0.012658  0.257307 -0.549500  0.096067  0.159651  0.216225   

                 Fare  
PassengerId  0.012658  
Survived     0.257307  
Pclass      -0.549500  
Age          0.096067  
SibSp        0.159651  
Parch        0.216225  
Fare         1.000000

> # .info()

**info()** fonksiyonu bize veri hakkında genel bilgiler verir. Verinin türü nedir? object, integer, or float? Kaç tane sütunlar da satır dolu?


**info()** fonksiyonu bize bunların cevabını verir.

### **Hadi örnek üstünde yapalım!**

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


> # isnull().sum()

**isnull().sum()**  burda iki tane gonksiyon var: **isnull()** ve **sum()**.
- **isnull()** fonksiyonu her satır için True veya False dödürür, eğer eksik veriyse True, değilse False.
- **sum()** fonksiyonu satırlardaki True değerlerini toplar bu sayede her sütunda kaç eksik veri var bunu görmüş oluruz.

### **Hadi örnek üstünde yapalım!**

In [10]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

> # .shape

**shape** fonksiyonu veride kaç satır ve kaç sütun olduğunu gösterir.

### **Hadi örnek üstünde yapalım!**

In [11]:
df.shape

(891, 12)

Verimizde 891 satır 12 sütun var.

---

![](https://miro.medium.com/max/1400/0*Tc3F4V9P2eBJBHlt)

[Unsplash](https://unsplash.com/?utm_source=medium&utm_medium=referral)'de [david Griffiths](https://unsplash.com/@itscakefortea?utm_source=medium&utm_medium=referral) tarafından bir fotoğraf

**EDA** sayesinde verilerden daha çok anlam çıkarabilceksiniz.


**Peki verilerin hepsini değil sadece ihtiyacımız olanı, sadece belli kısımlarını almak istersek ne yapmalıyız?**


Öğrenmek isterseniz :

# **Beklemede kalın!**
***"Veri filitreleme"*** yazımız gelicek!

---

# Ahmet Talha Bektaş
> 📧 [E-mail Hesabım](mailto:atalhabektas@gmail.com)

> 🔗 [LinkedIn Hesabım](https://www.linkedin.com/in/ahmet-talha-bekta%C5%9F-056844216)

> 💻 [GitHub Hesabım](https://github.com/ahmettalhabektastr)

> 👨‍💻 [Kaggle Hesabım](https://www.kaggle.com/ahmettalhabektastr)

> 📋 [Medium Hesabım](https://medium.com/@ahmettalhabektas_tr)